lib/c-api/wasmer_wasm.h

```diff
@@ -457,6 +497,88 @@ void wasm_module_name(const wasm_module_t *module, wasm_name_t *out);
  */
 bool wasm_module_set_name(wasm_module_t *module, const wasm_name_t *name);
 
+#if defined(WASMER_WASI_ENABLED)
+/**
+ * Non-standard function to get the module name of a
+ * `wasm_named_extern_t`.
+ */
+const wasm_name_t *wasm_named_extern_module(const wasm_named_extern_t *named_extern);
+#endif
+
+#if defined(WASMER_WASI_ENABLED)
+/**
+ * Non-standard function to get the name of a `wasm_named_extern_t`.
+ */
+const wasm_name_t *wasm_named_extern_name(const wasm_named_extern_t *named_extern);
+#endif
+
+#if defined(WASMER_WASI_ENABLED)
+/**
+ * Non-standard function to get the wrapped extern of a
+ * `wasm_named_extern_t`.
+ */
+const wasm_extern_t *wasm_named_extern_unwrap(const wasm_named_extern_t *named_extern);
+#endif
+
+#if defined(WASMER_WASI_ENABLED)
+/**
+ *Performs a deep copy of a vector of [`wasm_named_extern_t`].
+ */
+void wasm_named_extern_vec_copy(wasm_named_extern_vec_t *out_ptr,
+                                const wasm_named_extern_vec_t *in_ptr);
+#endif
+
+#if defined(WASMER_WASI_ENABLED)
+/**
+ *Deletes a vector of [`wasm_named_extern_t`].
+ *
+ *# Example
+ *
+ *See the [`wasm_named_extern_vec_t`] type to get an example.
+ */
+void wasm_named_extern_vec_delete(wasm_named_extern_vec_t *ptr);
+#endif
+
+#if defined(WASMER_WASI_ENABLED)
+/**
+ *Creates a new vector of [`wasm_named_extern_t`].
+ */
+void wasm_named_extern_vec_new(wasm_named_extern_vec_t *out,
+                               uintptr_t length,
+                               wasm_named_extern_t *const *init);
+#endif
+
+#if defined(WASMER_WASI_ENABLED)
+/**
+ *Creates an empty vector of [`wasm_named_extern_t`].
+ *
+ *# Example
+ *
+ *```rust # use inline_c::assert_c; # fn main() { #    (assert_c! { # #include "tests/wasmer_wasm.h" # int main() {     // Creates an empty vector of `wasm_named_extern_t`.     wasm_named_extern_vec_t vector;     wasm_named_extern_vec_new_empty(&vector);
```

Comment:
Do we usually include stuff like this in the generated header file? Seems like noise

Reply:
It is noise. It's from Rust comments, and automatically passed to C bindings with cbindgen. I'm not sure about how to solve that yet. We need those comments on the Rust side, but we don't want them inside the C headers.

Actually, we don't need any comments in the C headers at all. We have an online documentation, which is the best place where everything is documented. I'll try to see if we can remove comments in C headers with cbindgen.

Reply:
I've created a PR on cbindgen to skip documentation generation, https://github.com/eqrion/cbindgen/pull/652.

Reply:
PR to remove the documentation, https://github.com/wasmerio/wasmer/pull/2083